providers/implementations/digests/blake2_prov.c

```diff
@@ -83,23 +84,32 @@ static int blake2b512_internal_final(void *ctx, unsigned char *out,
                                      size_t *outl, size_t outsz)
 {
     struct blake2b_md_data_st *b_ctx;
-    
+
     b_ctx = (struct blake2b_md_data_st *)ctx;
-    *outl = b_ctx->ctx.outlen;
 
     if (!ossl_prov_is_running())
         return 0;
 
-    return (outsz > 0) ? ossl_blake2b_final(out, ctx) : 1;
+    *outl = b_ctx->ctx.outlen;
+
+    if (outsz == 0)
+       return 1;
+
+    if (outsz != *outl) {
```

Comment:
Shouldn't this really be `outsz < *outl`? In the future I might envisage some `EVP_DigestFinal_ex2` API that allows specifying the size of the `md` buffer, which could be larger than the minimum required.

Reply:
OK, fixed up